### PR TITLE
Clang core

### PR DESCRIPTION
**Pull request by den-mentiei**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,6 +10,8 @@ build/
 *.exe
 *.pdb
 *.ilk
+*.d
```
Comment (jimon) [inline .gitignore:13]: hm, ninja should have cleaned all of them if you have `deps = gcc`, so is it still needed ?

File: buildfox.py [hunk 2]
```diff
@@ -136,6 +136,98 @@
...
+		expand = true
+
+	rule link
+		command = $cxx $ldflags $libdirs $in -o $out $libs
+		description = link $out
+
+	rule link_so
+		command = $cxx -shared -fPIC $ldflags $libdirs -o $out $in $libs
+		description = cxx $in
+		expand = true
```
Comment (jimon) [inline buildfox.py]: Why we need expand for shared lib ?

File: .gitignore (hunk 1 above)
Comment (den-mentiei) [inline .gitignore:13]: It seems they are not always cleaned when ninja fails with some cxx errors.

File: buildfox.py [hunk 3]
```diff
@@ -136,6 +136,98 @@
...
+	# miscellaneous
+	cxx_fatal_warnings = -Werror
+	cxx_extra_warnings = -Wall -Wextra
+	cxx_no_warnings = -w
+
+	# transformers
+	defines =
+	includedirs =
+	libdirs =
+	libs =
```
Comment (jimon) [inline buildfox.py:216]: Looks like we need to have libs for msvc as well.

File: examples/shared_lib/withapp/build.fox [hunk 4]
```diff
@@ -12,4 +11,4 @@ build $out/test2/obj(*): auto test2/*.cpp
 build $out/shlib(test2) | $out/lib(test2): auto $out/test2/obj(*)
 
 build $out/obj(*): auto *.cpp
-build $out/app(app): auto $out/obj(*) $out/lib(*)
+build $out/app(app): auto $out/obj(*) $out/shlib(*)
```
Comment (jimon) [inline examples/shared_lib/withapp/build.fox]: Thats one tricky, looks like we link with .so on linux and with .lib on windows ... :/

Comment (den-mentiei) [inline examples/shared_lib/withapp/build.fox]: :rage4:

Comment (den-mentiei) [inline examples/shared_lib/withapp/build.fox]: That's the reason tests have failed.

**Pull request merged**